packages/consensus-validator-set-static/source/validator-set.ts

```diff
@@ -0,0 +1,42 @@
+import { inject, injectable, tagged } from "@mainsail/container";
+import { Contracts, Identifiers } from "@mainsail/contracts";
+import { Services } from "@mainsail/kernel";
+
+@injectable()
+export class ValidatorSet implements Contracts.Consensus.IValidatorSet {
+	@inject(Identifiers.Application)
+	private readonly app!: Contracts.Kernel.Application;
+
+	@inject(Identifiers.WalletRepository)
+	@tagged("state", "blockchain")
+	private readonly walletRepository!: Contracts.State.WalletRepository;
+
+	@inject(Identifiers.Cryptography.Identity.KeyPairFactory)
+	private readonly walletKeyPairFactory!: Contracts.Crypto.IKeyPairFactory;
+
+	// TODO: use PublicKeyFactory when possible
```

Comment:
Is there any particular reason why PublicKeyFactory wasn't used?

Reply:
We will solve it using tagged values.

Reply:
yeah, it requires tagging since it would use the wallet `KeyPairFactory`. as discussed this will be resolved in a follow up PR